doc/src/site/sphinx/ml/sw_glm.rst

```diff
@@ -0,0 +1,151 @@
+Train GLM Model in Sparkling Water
+----------------------------------
+Introduction
+~~~~~~~~~~~~
+
+Generalized Linear Models (GLM) estimate regression models for outcomes following exponential distributions.
+In addition to the Gaussian (i.e. normal) distribution, these include Poisson, binomial, and gamma distributions.
+Each serves a different purpose, and depending on distribution and link function choice, can be used either for prediction or classification.
+For more more comprehensive description see `H2O-3 GLM documentation <https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/glm.html>`__.
+
+Example
+~~~~~~~
+
+The following section describes how to train the GLM model in Sparkling Water in Scala & Python following the same example as H2O-3 documentation mentioned above. See also :ref:`parameters_H2OGLM` and :ref:`model_details_H2OGLMMOJOModel`.
+
+.. content-tabs::
+
+    .. tab-container:: Scala
+        :title: Scala
+
+        First, let's start Sparkling Shell as
+
+        .. code:: shell
+
+            ./bin/sparkling-shell
+
+        Start H2O cluster inside the Spark environment
+
+        .. code:: scala
+
+            import ai.h2o.sparkling._
+            import java.net.URI
+            val hc = H2OContext.getOrCreate()
+
+        Parse the data using H2O and convert them to Spark Frame
+
+        .. code:: scala
+
+            import org.apache.spark.SparkFiles
+            val datasetUrl = "https://raw.githubusercontent.com/h2oai/sparkling-water/master/examples/smalldata/prostate/prostate.csv"
+            spark.sparkContext.addFile(datasetUrl) //for example purposes, on a real cluster it's better to load directly from distributed storage
+            val rawSparkDF = spark.read.option("header", "true").option("inferSchema", "true").csv(SparkFiles.get("prostate.csv"))
+            val sparkDF = rawSparkDF.withColumn("CAPSULE", $"CAPSULE" cast "string")
```

Comment:
just FYI: you can also define the conversion by calling the `setColumnsToCategorical` method on the algorithm instance.